src/se/vidstige/jadb/JadbDevice.java

```diff
@@ -139,6 +141,30 @@ private StringBuilder buildCmdLine(String command, String... args) {
         return shellLine;
     }
 
+    /**
+     * Enable tcpip on the default port (5555)
+     *
+     * @return success or failure
+     */
+    public boolean enableTcpip() throws IOException, JadbException {
+        return enableTcpip(DEFAULT_TCPIP_PORT);
+    }
+
+    /**
+     * Enable tcpip on a specific port
+     *
+     * @param port for the device to bind on
+     *
+     * @return success or failure
+     */
+    public boolean enableTcpip(int port) throws IOException, JadbException {
+        try (Transport transport = getTransport()) {
+            send(transport, String.format("tcpip:%d", port));
+            String expectedResult = String.format("restarting in TCP Mode: %d\n", port);
+            return transport.readString(expectedResult.length()).equals(expectedResult);
+        }
```

Comment:
Hmm, this does not seem right either... 🤔 Do we have to parse the output of this? If the "send". The adb server should respond with a `"OKAY"` any way? E.g. the `send` function will already verify the "OKAY" response. I think we should make this function `void` instead, and just ignore the response string.